.github/actions/build-test/unix/action.yml

```diff
@@ -0,0 +1,67 @@
+name: Build & Test - Unix
+
+inputs:
+  GH_TOKEN:
+    required: true
+  BOOST_VERSION:
+    required: true
+  THRIFT_VERSION:
+    required: true
+  WARN_AS_ERR:
+    required: true
+  BUILD_TYPE:
+    required: true
+  SHARED_LIBS_TOGGLE:
+    required: true
+  OPENSSL_TOGGLE:
+    required: true
+  RUN_TESTS:
+    required: true
+  HAZELCAST_ENTERPRISE_KEY:
+    required: true
+  AWS_ACCESS_KEY_ID:
+    required: true
+  AWS_SECRET_ACCESS_KEY:
+    required: true
+  HZ_TEST_AWS_INSTANCE_PRIVATE_IP:
+    required: true
+
+env:
+  # Not possible to set this as a default
+  # https://github.com/orgs/community/discussions/46670
+  shell: bash
+
+runs:
+  using: composite
+  steps:
+    - name: Download hazelcast-enterprise-tests.jar
+      shell: ${{ env.shell }}
+      run: |
+        curl -H "Authorization: token ${{ inputs.GH_TOKEN }}" https://raw.githubusercontent.com/hazelcast/private-test-artifacts/data/certs.jar > hazelcast-enterprise-${{ env.HAZELCAST_VERSION }}-tests.jar
+
+    - name: Build & Install
+      env:
+        BUILD_DIR: build
+        INSTALL: ON
+        WARN_AS_ERR: ${{ inputs.WARN_AS_ERR }}
+      shell: ${{ env.shell }}
+      run: |
+        ./scripts/build-unix.sh                                          \
+            -DCMAKE_BUILD_TYPE=${{ inputs.BUILD_TYPE }}                  \
+            -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}/destination   \
+            -DBUILD_SHARED_LIBS=${{ inputs.SHARED_LIBS_TOGGLE }}         \
+            -DWITH_OPENSSL=${{ inputs.OPENSSL_TOGGLE }}                  \
+            -DBUILD_TESTS=ON                                             \
+            -DBUILD_EXAMPLES=OFF
+
+    - name: Test
```

Comment:
nit - `- name: Run test`

Reply:
> nit - `- name: Run test`

The rest are all named "Test" or "Test Memership" so I'd rather leave alone.